src/rbd.cc

```diff
@@ -1368,6 +1368,56 @@ static void update_snap_name(char *imgname, char **snap)
     *snap = s;
 }
 
+static bool is_valid_name(const char *name)
+{
+   char ch;
+   unsigned int i; 
+
+   if(name == NULL)
+     return true;
+
+   if(strlen(name) == 0)
+     return false;
+   
+   for (i = 0; i < strlen(name); i++ ) {
+      ch = name[i];
+      if ( !( ( ch >= '0' && ch <= '9' ) ||
```

Comment:
I'm not sure we should restrict names so much. It would be confusing if rbd had extra restrictions on the pool name, beyond the rados-enforced non-emptiness. I'm not sure we should define a character set at all, rather than behaving more like a fs and allowing any nonempty name not containing '\0'. Then any utf-8 string works fine.